src/applets/caffeine/CaffeineWindow.vala

```diff
@@ -0,0 +1,188 @@
+    private int32? default_brightness;
+
+    public AppletWindow (Gtk.EventBox? event_box, Settings? settings)
+    {
+        Object();
+        this.event_box = event_box;
+        this.settings = settings;
+
+        // Get settings
+        power_settings = new Settings (POWER_SCHEME);
+        session_settings = new Settings (SESSION_SCHEME);
+        try {
+            props = Bus.get_proxy_sync (BusType.SESSION,
+                                        "org.gnome.SettingsDaemon.Power",
+                                        "/org/gnome/SettingsDaemon/Power");
+        } catch (IOError e) {
+            print ("Error: %s\n", e.message);
+        }
+        fetch_default ();
+
+        mode.notify["active"].connect (on_mode_active);
+    }
+
+    public void toggle_applet ()
+    {
+        if (mode.active) {
+            mode.active = false;
+            return;
+        }
+        mode.active = true;
+    }
+
+    private void fetch_default ()
+    {
+        // Fetch default configuration
+        default_idle_delay = session_settings.get_uint ("idle-delay");
+        default_idle_dim = power_settings.get_boolean ("idle-dim");
+        default_sleep_inactive_ac_type = power_settings.get_string ("sleep-inactive-ac-type");
+        default_sleep_inactive_battery_type = power_settings.get_string ("sleep-inactive-battery-type");
+        try {
+            default_brightness = props.brightness;
+        } catch (DBusError e) {
+            print ("Error: %s\n", e.message);
+        }
+    }
+
+    private void change_brightness (int32 value)
+    {
+        try {
+            props.brightness = value;
+        } catch (DBusError e) {
+            print ("Error: %s\n", e.message);
+        }
+    }
+
+    private void send_notification (bool activate, int time)
+    {
+        var cmd = new StringBuilder ();
+        cmd.append ("notify-send ");
+
+        if (activate) {
+            cmd.append ("\"%s\" ".printf (_("Turn on Caffeine Boost")));
+
+            if (time > 0) {
+                var duration = ngettext ("a minute", "%d minutes", time).printf (time);
+                cmd.append ("\"%s %s\" ".printf (_("Will turn off in"), duration));
+            }
+            cmd.append ("--icon=caffeine-cup-full");
+        } else {
+            cmd.append ("\"%s\" ".printf (_("Turn off Caffeine Boost")));
+            cmd.append ("--icon=caffeine-cup-empty");
+        }
+
+        try {
+            Process.spawn_command_line_async (cmd.str);
+        } catch (SpawnError e) {
+            print ("Error: %s\n", e.message);
+        }
+    }
+
+    private void on_mode_active (Object? obj, ParamSpec? params)
+    {
+        var icon = event_box.get_child ();
+        event_box.remove (icon);
+
+        var time = timer.get_value_as_int();
+        if (mode.active)
+        {
+            // Fetch power settings default value
+            fetch_default ();
+
+            timer.sensitive = false;
```

Comment:
` timer.sensitive` setting should just be an inverse value of `mode.active` and be outside the if/else block IMO